lowfi_simulation/src/main/native/include/MotorEncoderSimulator.h

```diff
@@ -0,0 +1,34 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <memory>
+
+#include "EncoderSimulator.h"
+#include "MotorSimulator.h"
+
+namespace frc {
+namespace sim {
+namespace lowfi {
+
+class MotorEncoderSimulator {
+ public:
+  MotorEncoderSimulator(const std::shared_ptr<MotorSimulator>& motorController,
+                        const std::shared_ptr<EncoderSimulator>& encoder);
+  ~MotorEncoderSimulator();
+
+  void Update();
+
+ protected:
+  std::shared_ptr<MotorSimulator> motorSimulator;
```

Comment:
Why are these protected instead of private?

Reply:
I could see the potential of people overriding this and doing their own stuff with the interaction, but that might be premature optimization.  I'll change it to match the rest of the coding style